src/AnalyticsEventReporter.ts

```diff
@@ -0,0 +1,36 @@
+import { AnalyticsEventService } from "./AnalyticsEventService"
+import { AnalyticsConfig } from "./AnalyticsConfig";
+import { EventPayload } from "./EventPayload";
+import { EventAPIResponse } from "./EventAPIResponse";
+
+/** Represents an reporter is responsible for reporting analytics events. */
+export class AnalyticsEventReporter implements AnalyticsEventService {
+
+    private config: AnalyticsConfig;
+    private payload?: EventPayload;
+
+    /**
+     * @param config - necessary analytics config: Must provide one and only
+     * one of API Key or Bearer Token.
+     * 
+     * @param payload - (optional) desired event values to report
+     */
+    constructor(config: AnalyticsConfig, payload?: EventPayload) {
+        const apiKeyIsSet = typeof config.key === 'string' && config.key.trim().length !== 0;
+        const bearerTokenIsSet = typeof config.bearer === 'string' && config.bearer.trim().length !== 0;
+        const configIsValid = (apiKeyIsSet || bearerTokenIsSet) && !(apiKeyIsSet && bearerTokenIsSet);
```

Comment:
could just be apiKeyIsSet ^ bearerTokenIsSet right? (xor operator)

Reply:
Typescript does not support ^ for boolean values. I can do apiKeyIsSet !== bearerTokenIsSet, but think its readability is bad.

Reply:
^ is a bitwise operator. You can consider converting the boolean into a number, but that also might be a bit awkward for readability.